test/unit/serialization/compact/CompactTest.js

```diff
@@ -244,4 +246,19 @@ describe('CompactTest', function () {
         const object = await serializationService.toObject(data);
         object.should.be.deep.equal(employer);
     });
+
+    it('should throw proper error when nested field serializer is missing', async function() {
+        const bundle = createSerializationService(
+            [new MainDTOSerializer(), new InnerDTOSerializer()]
+        );
+
+        serializationService = bundle.serializationService;
+
+        const mainDTO = createMainDTO();
+        await expect(serialize(
+            serializationService,
+            bundle.schemaService,
+            mainDTO)).to.be.eventually.rejectedWith(HazelcastSerializationError).and.have.property('message',
+            'No serializer is registered for class/constructor NamedDTO.');
```

Comment:
```suggestion
            'No serializer is registered for class/constructor.');
```

An optional suggestion to save this test from failing when NamedDTO is renamed. Asserting whole message makes this test very easy to fail. Asserting   `No serializer is registered for class/constructor` part is enough to make sure correct HazelcastSerializationError is thrown and not another HazelcastSerializationError that might happen due to various reasons.